src/new-components/base-button/base-button.module.css

```diff
@@ -89,6 +83,26 @@
     transition-timing-function: cubic-bezier(0.4, 0, 0.2, 1);
 }
 
+.label {
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+    width: 100%;
```

Comment:
@gnapse I believe this will introduce a breaking change to the Twist login page when Reactist is updated to this version:
| Before | After |
| --- | --- |
| <img width="386" alt="image" src="https://user-images.githubusercontent.com/1051164/168882438-2fcc424e-8fb2-4e32-9f32-c1a074593f43.png"> | <img width="386" alt="image" src="https://user-images.githubusercontent.com/1051164/168882514-347088f0-0853-4050-a29d-cdaf0a4b11d7.png"> |

The only way around it seems to be adding custom CSS. Would it make sense for `width: 100%;` to only be applied when width is `full`?

Reply:
Indeed, if that fixes it, it makes sense that the label does not need to stretch in all situations. Does that fix it?

Reply:
Yeah it seems to fix it. I'll create a PR for it!